src/bigint.rs

```diff
@@ -66,6 +66,24 @@ impl Mul<Sign> for Sign {
     }
 }
 
+impl<T: PrimInt + BitAnd<Output=T>> Pow<T> for Sign {
+    type Output = Sign;
+
+    #[inline]
+    fn pow(self, other: T) -> Sign {
+        if self != Minus {
+            self
+        } else if other == T::zero() {
+            Plus
+        } else if other & T::one() == T::one() {
```

Comment:
Slightly better to use (other & T::one()).is_one() -- or if you switch to `T: Integer`, it's just `other.is_odd()`.